solutions/IOI 04-empodia.mdx

```diff
@@ -2,11 +2,13 @@
 id: ioi-04-empodia
 source: IOI 2004
 title: Empodia
-author: Benjamin Qi
+author: Benjamin Qi, Andi Qu
 ---
 
 [Official Editorial](https://ioinformatics.org/page/ioi-2004/30)
 
+**Time Complexity:** $\mathcal O(N)$.
```

Comment:
your implementation isn't O(N)? Also, no period

Reply:
Oops should I change the map into an unordered_map? If not, should I just use @tmwilliamlin168's implementation? (https://github.com/tmwilliamlin168/CompetitiveProgramming/blob/master/IOI/04-Empodia.cpp)

Reply:
wasn't my solution ok?

Reply:
The only issue with your solution was that it used templates, but since it's an IOI problem, that's probably ok to have. I've reverted back to your code (with the new CodeSnip feature)